### PR TITLE
Use `Node16` module resolution for typescript

### PR DESCRIPTION
**Pull request by Page-**
This continues to expose only commonjs but makes it decided by more standard nodejs means

Change-type: minor

### CODE REVIEW

**Comment by joshbwlng** [comment]
In order to get `npm run lint` and `npm run build` to succeed I had to also update imports from the following:
```
modified:   Gruntfile.ts
modified:   test/00-basic.test.ts
modified:   test/01-constrain.test.ts
modified:   test/02-sync-migrator.test.ts
modified:   test/03-async-migrator.test.ts
modified:   test/05-request-cancellation.test.ts
modified:   test/06-webresource.test.ts
modified:   test/07-permissions.test.ts
modified:   test/fixtures/04-translations/translations/hooks.ts
modified:   test/fixtures/05-request-cancellation/routes.ts
modified:   test/fixtures/06-webresource/translations/hooks.ts
modified:   test/lib/pine-init.ts
```

Also, for `Gruntfile.ts` I had to do this for `npm run build` to execute:
```
-export = (grunt: typeof Grunt) => {
+module.exports = (grunt: typeof Grunt) => {
```

File: src/database-layer/db.ts [hunk 1]
```diff
@@ -354,23 +354,22 @@ export abstract class Tx {
...
-		} finally {
-			this.listeners.rollback.forEach(asyncTryFn);
-			this.on = onRollback;
-			this.clearListeners();
-		}
+		this.listeners.rollback.forEach(asyncTryFn);
+		this.on = onRollback;
+		this.clearListeners();
+
+		await promise;
```
Comment (joshbwlng) [inline src/database-layer/db.ts:364]: Did you change the order of execution on purpose, now executing `await promise;` after what was originally in the `finally`?

File: src/sbvr-api/express-extension.ts [hunk 2]
```diff
@@ -1,15 +1,17 @@
 // Augment express.js with pinejs-specific attributes via declaration merging.
 
-// eslint-disable-next-line @typescript-eslint/no-namespace, @typescript-eslint/no-unused-vars
-declare namespace Express {
-	type PineUser = import('./sbvr-utils').User;
+declare global {
+	// eslint-disable-next-line @typescript-eslint/no-namespace
+	namespace Express {
+		type PineUser = import('./sbvr-utils').User;
```
Comment (joshbwlng) [inline src/sbvr-api/express-extension.ts:6]:
```suggestion
		type PineUser = import('./sbvr-utils.js').User;
```

File: src/sbvr-api/express-extension.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+		// Augment Express.User to include the props of our PineUser.
+		// eslint-disable-next-line @typescript-eslint/no-empty-interface
+		interface User extends PineUser {}
 
-	interface Request {
-		user?: User;
-		apiKey?: import('./sbvr-utils').ApiKey;
+		interface Request {
+			user?: User;
+			apiKey?: import('./sbvr-utils').ApiKey;
```
Comment (joshbwlng) [inline src/sbvr-api/express-extension.ts:14]:
```suggestion
			apiKey?: import('./sbvr-utils.js').ApiKey;
```

File: src/database-layer/db.ts (hunk 1 above)
Comment (Page-) [reply src/database-layer/db.ts]: These changes snuck in by accident - I've removed them now. Thanks for catching

**Review by thgreasi — APPROVED** [review]

File: src/server-glue/module.ts [hunk 4]
```diff
@@ -66,7 +66,7 @@ export const init = async <T extends string>(
 
 		const promises: Array<Promise<void>> = [];
 		if (process.env.SBVR_SERVER_ENABLED) {
-			const sbvrServer = await import('../data-server/sbvr-server');
+			const sbvrServer = await import('../data-server/sbvr-server.js');
```
[the remark this answers is not in the document]
Comment (Page-) [reply src/server-glue/module.ts:69]: Yeah as long as we're running node16+ (which is the minimum we support) then it's fine

**Pull request merged**